mlx_lm/server.py

```diff
@@ -532,12 +532,15 @@ def _generate(self):
 
         unprocessed_requests = []
 
-        def get_next_request():
+        def get_next_request(timeout=None):
             if unprocessed_requests:
                 return unprocessed_requests.pop()
             else:
                 try:
-                    return self.requests.get_nowait()
+                    if timeout is not None:
+                        return self.requests.get(timeout=timeout)
```

Comment:
Why use a timeout? Why not just block until the queue has a request in this case?